src/types.rs

```diff
@@ -1,5 +1,45 @@
-pub use massa_hash::Hash;
-pub use massa_signature::KeyPair;
-pub use massa_signature::PublicKey;
-pub use massa_signature::Signature;
-pub const PUBLIC_KEY_SIZE_BYTES: usize = massa_signature::PUBLIC_KEY_SIZE_BYTES;
+use std::fmt::Debug;
+use std::fmt::Display;
+use std::hash::Hash;
+
+use crate::error::PeerNetResult;
+
+// pub const PUBLIC_KEY_SIZE_BYTES: usize = massa_signature::PUBLIC_KEY_SIZE_BYTES;
+
+pub trait PeerNetKeyPair<K>: Send + Sync + Clone + Debug + Display + 'static {
```

Comment:
Is that possible to constraint K to implement `PeerNetPubKey` ?

Reply:
yes i will add it

Reply:
yes, done